src/abstractdataframe/reshape.jl

```diff
@@ -399,3 +399,20 @@ function CategoricalArrays.CategoricalArray(v::RepeatedVector)
     res.refs = repeat(res.refs, inner = [v.inner], outer = [v.outer])
     res
 end
+
+
+transpose1(df::AbstractDataFrame, indexcol=1) = unstack(stack(df,Not(indexcol)), :variable, indexcol,:value)
```

Comment:
in general - the output index column name should be passed independently + some check if it conflicts with the generated column names should be made.

Reply:
Agree - was thinking it might be worth putting the `makeunique` keyword in there to deal with cases where the index column is not all unique.

Reply:
yes, this is a standard we do 😄.